lib/sablon/parser/mail_merge.rb

```diff
@@ -108,9 +109,30 @@ def replace(content, env)
           end
 
           unless env.remove_fields_only
-            replace_field_display(pattern_node, content, env)
-            (@nodes - [pattern_node]).each(&:remove) unless env.keep_merge_fields
+            the_pattern_node = pattern_node(!inserting_word_ml)
+            replace_field_display(the_pattern_node, content, env)
+            if env.keep_merge_fields
```

Comment:
which `env` is this referring to? @fp52oxe   Not the ssd app env?

Reply:
@rpfondeker It's their internal collection of options